### PR TITLE
fix(server): Enrich user messages with Kind and MessageID fields when creating tasks

### LINKED ISSUE #76: Server should enrich user messages with Kind and MessageID fields when creating tasks
## Bug Description

When a client sends a message via `message/send` without `Kind` and `MessageID` fields set, the server stores the message as-is in the task history. This results in user messages having empty `kind` and `messageId` fields in the task history, while assistant messages (generated by task handlers) have proper values.

## Expected Behavior

All messages in task history should have proper `Kind` and `MessageID` fields, regardless of whether the client provides them or not.

## Current Behavior

- User messages from clients: `"kind": "", "messageId": ""`
- Assistant messages from server: `"kind": "message", "messageId": "<uuid>"`

## Example Output

```json
{
  "history": [
    {
      "kind": "",           // ❌ Empty
      "messageId": "",      // ❌ Empty  
      "role": "user",
      "parts": [...]
    },
    {
      "kind": "message",    // ✅ Correct
      "messageId": "uuid",  // ✅ Correct
      "role": "assistant", 
      "parts": [...]
    }
  ]
}
```

## Root Cause

In `server/task_handler.go`, the `createTaskFromMessage` function passes `&params.Message` directly to `CreateTask` without enriching the message with required fields.

## Proposed Solution

Enrich incoming messages in `createTaskFromMessage` before creating the task:

```go
// Enrich the message with Kind and MessageID if not set
enrichedMessage := params.Message
if enrichedMessage.Kind == "" {
    enrichedMessage.Kind = "message"
}
if enrichedMessage.MessageID == "" {
    enrichedMessage.MessageID = uuid.New().String()
}
```

## Impact

- Ensures consistent message format in task history
- Maintains backward compatibility with existing clients
- Follows the pattern already used for assistant messages

## Files to Change

- `server/task_handler.go` - Add message enrichment in `createTaskFromMessage` function

## Test Cases

- Client sends message without `Kind` and `MessageID` → Server should populate them
- Client sends message with `Kind` and `MessageID` → Server should preserve them
- All messages in task history should have consistent format

### PR DESCRIPTION
**Pull request by edenreich**
Fixes #76

This PR addresses the bug where user messages from clients had empty `Kind` and `MessageID` fields in task history, while assistant messages had proper values.

## Changes
- Add message enrichment logic in `CreateTaskFromMessage` function
- Ensure all messages in task history have consistent `Kind` and `MessageID` fields
- Maintain backward compatibility by preserving existing values
- Add comprehensive test suite with 5 test cases covering edge cases
- Make `CreateTaskFromMessage` public for testability

## Test Coverage
- Message without Kind and MessageID → Gets enriched
- Message with empty fields → Gets enriched
- Message with existing values → Values preserved
- Partial field scenarios → Only missing fields enriched

Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: server/task_handler.go [hunk 1]
```diff
@@ -677,16 +677,25 @@ func NewDefaultA2AProtocolHandler(
...
 
-// createTaskFromMessage creates a task directly from message parameters
-func (h *DefaultA2AProtocolHandler) createTaskFromMessage(ctx context.Context, params types.MessageSendParams) (*types.Task, error) {
+// CreateTaskFromMessage creates a task directly from message parameters
+func (h *DefaultA2AProtocolHandler) CreateTaskFromMessage(ctx context.Context, params types.MessageSendParams) (*types.Task, error) {
 	if len(params.Message.Parts) == 0 {
 		return nil, fmt.Errorf("empty message parts not allowed")
 	}
 
+	// Enrich the message with Kind and MessageID if not set
```
Comment (edenreich) [inline server/task_handler.go:686]:
```suggestion
```

File: server/task_handler_test.go [hunk 2]
```diff
@@ -446,6 +446,157 @@ func TestDefaultA2AProtocolHandler_ContextHistoryHandling(t *testing.T) {
...
+			assert.NotNil(t, task)
+
+			// Verify that CreateTask was called with the enriched message
+			assert.Equal(t, 1, mockTaskManager.CreateTaskCallCount())
+			_, _, enrichedMessage := mockTaskManager.CreateTaskArgsForCall(0)
+
+			// Check that Kind was enriched correctly
+			assert.Equal(t, tt.expectedKind, enrichedMessage.Kind)
+
+			// Check MessageID enrichment
```
Comment (edenreich) [inline server/task_handler_test.go:585]:
```suggestion
```

File: server/task_handler_test.go [hunk 3]
```diff
@@ -446,6 +446,157 @@ func TestDefaultA2AProtocolHandler_ContextHistoryHandling(t *testing.T) {
...
+			task, err := handler.CreateTaskFromMessage(ctx, params)
+
+			assert.NoError(t, err)
+			assert.NotNil(t, task)
+
+			// Verify that CreateTask was called with the enriched message
+			assert.Equal(t, 1, mockTaskManager.CreateTaskCallCount())
+			_, _, enrichedMessage := mockTaskManager.CreateTaskArgsForCall(0)
+
+			// Check that Kind was enriched correctly
```
Comment (edenreich) [inline server/task_handler_test.go:582]:
```suggestion
```

File: server/task_handler_test.go [hunk 4]
```diff
@@ -446,6 +446,157 @@ func TestDefaultA2AProtocolHandler_ContextHistoryHandling(t *testing.T) {
...
+			}
+			params.Message.ContextID = &contextID
+
+			ctx := context.Background()
+			task, err := handler.CreateTaskFromMessage(ctx, params)
+
+			assert.NoError(t, err)
+			assert.NotNil(t, task)
+
+			// Verify that CreateTask was called with the enriched message
```
Comment (edenreich) [inline server/task_handler_test.go:578]:
```suggestion
```
